api/reporting/schemas/reporting-codegen.rb

```diff
@@ -136,68 +261,190 @@ def to_swagger_property(column)
-    'INT64' => 'long',
-    'TIMESTAMP' =>  'Timestamp',
-    'BOOLEAN' =>  'boolean',
-    'FLOAT64' => 'double'
-}
+write_output(OUTPUTS[:swagger_yaml], indented_yaml, 'DTO Swagger Definition')
 
+### Projection Interface
 def to_getter(field)
-  "  #{BIGQUERY_TYPE_TO_JAVA[field[:big_query_type]]} get#{to_camel_case(field[:name], true)}();"
+  property_type = field[:java_type]
+  "  #{property_type} #{field[:getter]}();"
 end
 
-getters = columns.map { |field|
+projection_decl = "public interface #{TABLE_INFO[:projection_interface]} {\n"
+projection_decl << COLUMNS.map { |field|
   to_getter(field)
-}
-
-def projection_name(table_name)
-  "Prj#{to_camel_case(table_name, true )}"
-end
-
-java = "public interface #{projection_name(table_name)} {\n"
-java << getters.join("\n")
-java << "\n}\n"
+}.join("\n")
+projection_decl << "\n}\n"
 
-IO.write(outputs[:projection_interface], java)
-puts "  Spring Data Projection Interface: #{outputs[:projection_interface]}"
+write_output(OUTPUTS[:projection_interface], projection_decl, 'Spring Data Projection Interface')
 
 ### Projection query
 def hibernate_column_name(field)
   to_camel_case(field[:name], false)
 end
 
 # Fix up research purpose entity fields, which don't match the column names (i.e. there's no 'rp' prefix)
-def adjust_rp_col(field, table_alias)
+def adjust_col(field)
   md = field.match(/^rp_(?<root>\w+$)/)
   projection_field = to_camel_case(field, false)
   entity_property = projection_field
   if md and md['root']
     entity_property = to_camel_case(md['root'], false)
-    "#{table_alias}.#{entity_property} AS #{projection_field}"
-  else
-    "#{table_alias}.#{entity_property}"
+    "#{TABLE_INFO[:sql_alias]}.#{entity_property} AS #{projection_field}"
+  elsif TABLE_INFO[:entity_modified_columns][field]
+     "#{TABLE_INFO[:sql_alias]}.#{TABLE_INFO[:entity_modified_columns][field]}"
+   else
+    "#{TABLE_INFO[:sql_alias]}.#{entity_property}"
   end
 end
 
-  def to_query(table_name, schema)
-  table_alias = table_name[0].downcase
+def to_query()
   "@Query(\"SELECT\\n\"\n" \
-    + schema.map do |field|
-      "+ \"  #{adjust_rp_col(field[:name], table_alias)}"
+    + COLUMNS.map do |field|
+      "+ \"  #{adjust_col(field[:name])}"
     end \
     .join(",\\n\"\n") \
     + "\\n\"\n" \
-    + "+ \"FROM Db#{to_camel_case(table_name, true)} #{table_alias}\")\n" \
-    + "  List<#{projection_name(table_name)}> getReporting#{to_camel_case(table_name, true)}s();"
+    + "+ \"FROM #{TABLE_INFO[:entity_class]} #{TABLE_INFO[:sql_alias]}\")\n" \
+    + "  List<#{TABLE_INFO[:projection_interface]}> getReporting#{to_camel_case(TABLE_INFO[:name], true)}s();"
+end
+
+sql = to_query
+
+write_output(OUTPUTS[:projection_query], sql, 'Projection Query')
+
+# Unit Test Constants
+#
+BASE_TIMESTAMP = Time.new(2015, 5, 5).freeze
+TIMESTAMP_DELTA_SECONDS = 24 * 60 * 60 # .freeze # seconds in day
+
+# N.B. some Short fields are only valid up to the number of associated enum values - 1. Fixing these
```

Comment:
for the sake of paranoia, I wanted to make sure almost now columns had the same value for a given row. It's very easy to get things switched around, and if you make an error like that, the preview text in the debugger will show keys and values all mismatched and scare you to death..